### ISSUE TITLE
Bug: label + for() statement crashing tab/client

### ISSUE THREAD

**Issue by tomsiwik** [codesandbox/codesandbox-client#1288]
# 🐛 bug report

Pretty thanks in looking into it, in advance.

## Description of the problem

Using `label:` and a `for()` statement crashes the sandbox. Used typescript + react template.

## How has this issue affected you? What are you trying to accomplish?

Using labels is not possible as per:
https://developer.mozilla.org/de/docs/Web/JavaScript/Reference/Statements/label

### Your Environment
| Software         | Name/Version|
| ---------------- | ---------- |
| Сodesandbox  | PROD-1541378538-cfd11b5
| Browser | Chrome
| Operating System | Mac

**Comment by SaraVieira** [comment]
It seems the sandbox link no longer has the code mentioned :/

**Comment by tomsiwik** [comment]
Apparently this issue now temporary crashes the full link. New link edited and found [here](https://codesandbox.io/s/r75pj4l16p).

```javascript
/* 
 * Reproduce with uncommenting below line
 */
// label:
for()
```

**Comment by CompuIves** [comment]
Hmm, well the browser is crashing because that's an infinite loop being executed. Since the sandbox preview shares the same thread as the editor it can freeze up the UI. We do have some precautions for this scenario:

1. We save code changes to localStorage, so you won't miss them
2. Next time you open the sandbox we don't execute the code so you can make the proper changes

We also have a babel plugin that throws an error after x iterations in a loop, but it seems like that doesn't catch the label+loop scenario.

I'm not sure if there's more that we can do against this, except for running the preview code in another thread, but that's more a browser functionality than something that we can do.

**Comment by tomsiwik** [comment]
Thank you for the response. I didn't know that the browser executes an empty `for()` without curly braces. You are right, apparently label isn't the culprit at all. 

In response to point 2: when I click next to the for loop and want to change something it immediately re-executes the code again. Isn't there some on demand method that can pause the auto-execution? The only way I can circumvent this issue by hiding the browser view.

Long story short, the explanation helped and fixed my problem. Thank you very much. Usability-wise, maybe a pause button or similar... or something that does not rebuild the code when the for loop isn't finished for fools like me that always start to write `for()`.
